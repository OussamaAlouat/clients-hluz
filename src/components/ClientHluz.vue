<template>
  <v-card class="mx-auto" max-width="344" outlined>
    <v-list-item three-line>
      <v-list-item-content>
        <v-list-item-title class="text-h5 mb-1">
          Cecil Grant
        </v-list-item-title>
        <v-list-item-subtitle>Sweet potato street, 2</v-list-item-subtitle>
        <v-list-text>customer-basic</v-list-text>
      </v-list-item-content>

      <v-list-item-avatar tile size="80" color="grey">Build</v-list-item-avatar>
    </v-list-item>
  </v-card>
</template>

<script>

export default {
  name: 'ClientHluz',
  props: {
    clientName: {
      type: String,
      required: true,
    },

    adress: {
      type: String,
      required: true,
    },

    role: {
      type: String,
      required: true,
    },
    buildingType: {
      type: String,
      required: true,
    },
  },
};
</script>

<style>

</style>
