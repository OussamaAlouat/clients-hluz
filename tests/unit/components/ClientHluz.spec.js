import { shallowMount } from '@vue/test-utils';
import ClientHluz from '@/components/ClientHluz.vue';

describe('ClientHluz.vue', () => {
  const props = {
    clientName: 'client',
    address: 'address1 ',
    role: 'role1',
    buildType: 'house',
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(ClientHluz, {
      propsData: { ...props },
    });
  });

  it('Component render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
