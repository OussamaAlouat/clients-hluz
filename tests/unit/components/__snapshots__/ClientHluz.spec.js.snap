// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientHluz.vue Component render 1`] = `
<div class="client-box" buildtype="house">
  <v-row-stub tag="div">
    <v-col-stub tag="div">
      Name: client
    </v-col-stub>
    <v-col-stub tag="div">
      Address: address1
    </v-col-stub>
  </v-row-stub>
  <v-row-stub tag="div">
    <v-col-stub tag="div">
      Role: role1
    </v-col-stub>
    <v-col-stub tag="div">
      Building type:
    </v-col-stub>
  </v-row-stub>
</div>
`;
